test/test_integration_credentials_keyring.py

```diff
@@ -333,7 +335,7 @@ def test_https_can_load_policy(self):
     def test_keyring_locked_in_the_middle(self):
```

Comment:
`test_keyring_locked_after_login_will_raise_error`

What are we checking here? That we get an error when logout? I think a couple of things. 1. add a comment what this test is doing and the purpose 2. A more interesting command to run is `list` to ensure we cannot run because of the keyring is locked. 3. What happens when the netrc has the same creds as the creds in the keyring? what should happen? Should we fail should we remove from netrc?